src/Playwright/Helpers/Paths.cs

```diff
@@ -34,18 +35,55 @@ internal static class Paths
         internal static string GetExecutablePath()
         {
             var assembly = typeof(Playwright).Assembly;
-            string driversPath = new FileInfo(assembly.Location).Directory.FullName;
+            string driversPath, executableFile;
 
-            string executableFile = GetPath(driversPath);
+            // in .NET 6, the notion of single-assembly apps means, we won't have a location
+            // and this can throw an exception - let's guard against it
+            if (!string.IsNullOrEmpty(assembly.Location))
+            {
+                driversPath = new FileInfo(assembly.Location).Directory.FullName;
+
+                executableFile = GetPath(driversPath);
+                if (File.Exists(executableFile))
+                {
+                    return executableFile;
+                }
+            }
+
+            // fallback to the NuGet repository
+            driversPath = GetNugetPackagePath(assembly.GetName().Version.ToString(3));
+            executableFile = GetPath(driversPath, "Drivers");
             if (File.Exists(executableFile))
             {
                 return executableFile;
             }
 
-            throw new PlaywrightException($@"Driver not found: {executableFile}");
+            throw new PlaywrightException($"Driver not found: {executableFile}");
+        }
+
+        private static string GetNugetPackagePath(string version)
+        {
+            string packagesPath = Environment.GetEnvironmentVariable("NUGET_PACKAGES")
+                ?? Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".nuget", "packages");
+
+            string packagePath = Path.Combine(packagesPath, "microsoft.playwright");
+            if (Directory.Exists(packagePath))
+            {
+                var path = Directory
+                    .GetDirectories(packagePath, version + "*")
+                    .OrderByDescending(x => x, new Naturally.NaturalSortOrderStringComparer())
```

Comment:
I don't think we should be doing this. In the case we are interested in, assembly itself belongs to nuget registry, so you just need to pick a corresponding driver.

Reply:
Completely wishlist, but at least for the linqpad case, typeof(Microsoft.Playwright.Playwright).Assembly.Location is going to give us the location in the extracted package of the assembly, and 2 up from there is the extracted package root.  Going based on NUGET_PACKAGES and ~/.nuget/packages seems more limited (and has to do this version logic) than just "support running from an extract nuget dir" which would work for any location on disk, although I understand the single-assembly apps means needing to fallback to nuget anyway.

I guess my preference would be a 
1) normal build output - .playwright dir at assembly location
2) extracted nuget package, works regardless of version or extracted location
3) nuget package, needed for single-assembly apps

Something like this (likely not exactly this, just to get the idea across)

```csharp
var driversDirectoryPathsToTry = new List<string>();

var assembly = typeof(Microsoft.Playwright.Playwright).Assembly;
// in .NET 6, the notion of single-assembly apps means, we won't have a location
// and this can throw an exception - let's guard against it
if (!string.IsNullOrEmpty(assembly.Location))
{
    var assemblyParentDir = Path.GetDirectoryName(assembly.Location);

    // normal build output
    driversDirectoryPathsToTry.Add(Path.Combine(assemblyParentDir, ".playwright"));
    // running from extracted nuget dir
    driversDirectoryPathsToTry.Add(Path.Combine(assemblyParentDir, "..", "..", "Drivers"));
}
var driversPathUnderNuget = GetDriversPathUnderNugetPackagePath();
if (!string.IsNullOrEmpty(driversPathUnderNuget))
{
    driversDirectoryPathsToTry.Add(driversPathUnderNuget);
}

foreach (var pathToTry in driversDirectoryPathsToTry)
{
    var runnerPath = Path.Combine(pathToTry, "node", platformId, runnerName);
    if (File.Exists(runnerPath))
    {
        return runnerPath;
    }
}
throw new PlaywrightException($"Microsoft.Playwright drivers not found, tried in paths: {String.Join(", ", driversDirectoryPathsToTry)}");
```

Then iterate over the list and return the first path where

Reply:
Yeah, that's what I've done now. I think this should cover your use case. As per @pavelfeldman's suggestion, we're not going to fix the single-file just yet, so no use in including that fix in here. Let's fix this first. :)